src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/DebugFuncEval.cs

```diff
@@ -0,0 +1,143 @@
+            IntPtr targetAddress = RuntimeImports.RhpGetFuncEvalTargetAddress();
+
+            // Hard coding a single void return here
+            LocalVariableType[] returnAndArgumentTypes = new LocalVariableType[2];
+            returnAndArgumentTypes[0] = new LocalVariableType(typeof(void).TypeHandle, false, false);
+            returnAndArgumentTypes[1] = new LocalVariableType(typeof(int).TypeHandle, false, false);
+
+            // 2) Invoke the target method
+
+            // Hard coding static here
+            DynamicCallSignature dynamicCallSignature = new DynamicCallSignature(Internal.Runtime.CallConverter.CallingConvention.ManagedStatic, returnAndArgumentTypes, returnAndArgumentTypes.Length);
+
+            // Go and execute!
+            Internal.Runtime.CallInterceptor.CallInterceptor.MakeDynamicCall(targetAddress, dynamicCallSignature, arguments);
+
+            FuncEvalCompleteCommand funcEvalCompleteCommand = new FuncEvalCompleteCommand();
+            funcEvalCompleteCommand.commandCode = 0;
+
+            using (AutoPinner ap = new AutoPinner(funcEvalCompleteCommand))
+            {
+                RuntimeImports.RhpSendCustomEventToDebugger(ap, Unsafe.SizeOf<FuncEvalCompleteCommand>());
+            }
+        }
+
+        [StructLayout(LayoutKind.Explicit, Size=16)]
+        struct WriteParameterCommand
+        {
+            [FieldOffset(0)]
+            public int commandCode;
+            [FieldOffset(4)]
+            public int unused;
+            [FieldOffset(8)]
+            public long bufferAddress;
+        }
+
+        [StructLayout(LayoutKind.Explicit, Size=4)]
+        struct FuncEvalCompleteCommand
+        {
+            [FieldOffset(0)]
+            public int commandCode;
+        }
+
+        private static void HighLevelDebugFuncEvalHelper()
+        {
+            int integerParameterValue = 0;
+
+            uint parameterBufferSize = RuntimeImports.RhpGetFuncEvalParameterBufferSize();
+            byte[] debuggerBuffer = new byte[parameterBufferSize];
+
+            using (AutoPinner ap = new AutoPinner(debuggerBuffer))
+            {
+                IntPtr debuggerBufferPointer = ap;
+
+                WriteParameterCommand writeParameterCommand = new WriteParameterCommand();
+                writeParameterCommand.commandCode = 1;
+                writeParameterCommand.bufferAddress = debuggerBufferPointer.ToInt64();
+                
+                using (AutoPinner ap2 = new AutoPinner(writeParameterCommand))
+                {
+                    RuntimeImports.RhpSendCustomEventToDebugger(ap2, Unsafe.SizeOf<WriteParameterCommand>());
+                }
+
+                unsafe
+                {
+                    integerParameterValue = Unsafe.Read<int>(debuggerBufferPointer.ToPointer());
+                }
+            }
+
+            // Hard coding a single argument of type int here
+            LocalVariableType[] argumentTypes = new LocalVariableType[2];
```

Comment:
Won't fix - don't worry about that, this is currently hard coded that will be replaced by a general signature parsing mechanism.